docs/sts-migrate-to-private-bucket.md

```diff
@@ -0,0 +1,491 @@
+
+> NOTE 1: from here, the cluster will lose access to the integrated components (machine-api, image registry, CSI, ...)
+
+> NOTE 2: The script below should be run carefully, it was created to show the current and desired policies. If you find anything that does not match the expected changes, abort it immediately.
+
+> Helper [`aws iam get-role`](https://docs.aws.amazon.com/cli/latest/reference/iam/get-role.html)
+
+> Helper [`aws iam update-assume-role-policy`](https://docs.aws.amazon.com/cli/latest/reference/iam/update-assume-role-policy.html)
+
+```bash
+for ROLE_NAME in $(jq -r .RoleName iam-roles-new.json);
+do
+    echo -e ">>>>>\n#> (1) CURRENT IAM Role \"$ROLE_NAME\":";
+    aws iam get-role --role-name $ROLE_NAME | jq .Role;
+
+    echo -e "\n#> (2) NEW IAM Role \"$ROLE_NAME\" AssumeRolePolicyDocument:";
+    jq -r ". | select(.RoleName == \"$ROLE_NAME\").AssumeRolePolicyDocument" iam-roles-new.json \
+        | tee ${PWD}/iam-roles-new-$ROLE_NAME.json
+    
+    read -p "ATTENTION: The AssumeRolePolicyDocument for IAM Role(1) will be patched to the value of (2). Do you want to continue? [y/n]: " answer
+    if [ -z "$answer" ] || [ "$answer" != "y" ]
+    then
+        echo "answer[$answer]. Canceling the operation.";
+        break
+    fi
+    echo "Patching..."
+    aws iam update-assume-role-policy \
+        --role-name $ROLE_NAME \
+        --policy-document file:///${PWD}/iam-roles-new-$ROLE_NAME.json
+    echo "Done! Return code=$?"
+done
+```
+
+- Patch the issuer URL to the new OIDC URL on the Authentication object:
+
+```bash
+oc patch authentication cluster \
+    --type=merge \
+    -p "{\"spec\":{\"serviceAccountIssuer\":\"https://${CLOUDFRONT_HOST}\"}}"
+```
+
+- Wait for the kube-apiserver rollout
+
+> Wait to clean the `PROGRESSING=TRUE`. It could take some minutes to start and complete.
+
+```bash
+$ oc get co kube-apiserver -w
+$ oc get pods -n openshift-kube-apiserver -l apiserver=true -w
+```
+
+- Restart all pods:
+
+```bash
+for I in $(oc get ns -o jsonpath='{range .items[*]} {.metadata.name}{"\n"} {end}'); \
+      do oc delete pods --all -n $I; \
+      sleep 1; \
+      done
+```
+
+- Test the new token
+
+> Repeat the steps in the section "Test the existing token"
+
+> Make sure the JWT token has the CloudFront Distribution Domain name as the Issuer URL, field `.iss`
+
+> Make sure you can assume the role correctly and the signer will be the CloudFront: `.Provider` in the answer from `assume-role-with-web-identity`
+
+If you have completed these steps successfully, the cluster is using the new identity provider with AWS CloudFront.
+
+### Revoke public access to the S3 Bucket<a name="revoke-s3-public-access"></a>
+
+- Change the default policy blocking public access to the bucket:
+
+```bash
+aws s3api put-public-access-block \
+    --bucket ${OIDC_BUCKET_NAME} \
+    --public-access-block-configuration \
+    BlockPublicAcls=true,IgnorePublicAcls=true,BlockPublicPolicy=true,RestrictPublicBuckets=true
```

Comment:
executed with user jshu-sts, reported the following error:
[cloud-user@preserve-for-hive-test testdir]$ aws s3api put-public-access-block     --bucket ${OIDC_BUCKET_NAME}     --public-access-block-configuration     BlockPublicAcls=false,IgnorePublicAcls=false,BlockPublicPolicy=false,RestrictPublicBuckets=false

An error occurred (AccessDenied) when calling the PutPublicAccessBlock operation: Access Denied

After switched back my admin user, it worked